### PR TITLE
Refactoring Application Update implementation

### PR DESCRIPTION
**Pull request by lawrab**
- moved from UI to a Plugin
- Moved init.lua text to a resource instead of string in code

### CODE REVIEW

File: Components/AppilcationUpdate/Plugins/ApplicationUpdatePlugin.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            SubscribeToInternalEvents();
+
+            if (Debugger.IsAttached)
+            {
+                Logger.Information("Auto update is disabled when a Debugger is attached");
+                return;
+            }
+
+            //var updateLocation = configuration.ExtractOrDefault("updateLocation", string.Empty); 
+            //var prerelease = configuration.ExtractOrDefault("prerelease", false);
```
Comment (dennis) [inline Components/AppilcationUpdate/Plugins/ApplicationUpdatePlugin.cs:56]: You can remove these - they are a few lines down as well

File: Components/AppilcationUpdate/Plugins/ApplicationUpdatePlugin.cs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        private void SubscribeToInternalEvents()
+        {
+            // Registering for internal plugin state event
+            var internalEvents = EventHandlerController.Get<Internal.EventHandler.Internal>();
+            internalEvents.OnInternalPluginState += (s, e) => OnInternalPluginState(s, e);
+        }
+
+        private void OnInternalPluginState(IEventHandlerController s, EventHandlerArgs<InternalPluginState> e)
+        {
+            // to prevent multiple updates
```
Comment (dennis) [inline Components/AppilcationUpdate/Plugins/ApplicationUpdatePlugin.cs:78]: You can refer to the https://github.com/dennis/slipstream/issues/84

File: Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -32,6 +32,6 @@
 // You can specify all the values or you can default the Build and Revision Numbers
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("0.4.1")]
-[assembly: AssemblyFileVersion("0.4.1")]
+[assembly: AssemblyVersion("0.4.2")]
```
Comment (dennis) [inline Properties/AssemblyInfo.cs:35]: Did you intent to do this? If so, we should update changelog and tag it.

File: Components/AppilcationUpdate/Plugins/ApplicationUpdatePlugin.cs (hunk 1 above)
Comment (lawrab) [reply Components/AppilcationUpdate/Plugins/ApplicationUpdatePlugin.cs:56]: Removed

File: Properties/AssemblyInfo.cs (hunk 3 above)
Comment (lawrab) [reply Properties/AssemblyInfo.cs:35]: This was only a test, I will create 0.5.0 once all the changes are reviewed and you are happy and I merged from master with your latest changes.

**Review by dennis — APPROVED** [review]

**Pull request merged**